kong/db/declarative/migrations/route_path.lua

```diff
@@ -4,29 +4,44 @@ local pairs  = pairs
 local ipairs = ipairs
 local null   = ngx.null
 
-return function(tbl, version)
-  if not tbl or not (version == "1.1" or version == "2.1") then
-    return
-  end
+local empty_tbl = {}
 
-  local routes = tbl.routes
+local function table_default(val)
+  -- we cannot verify it with tyoe(val) == "table" because
+  -- we may recieve indexable cdata/lightuserdata
+  if val and val ~= null then
+    return val
 
-  if not routes then
-    -- no need to migrate
-    return
+  else
+    return empty_tbl
   end
+end
 
+local function migrate_routes(routes)
   for _, route in pairs(routes) do
-    local paths = route.paths
-    if not paths or paths == null then
-      -- no need to migrate
-      goto continue
-    end
+    local paths = table_default(route.paths)
 
     for idx, path in ipairs(paths) do
       paths[idx] = migrate_path(path)
     end
+  end
+end
+
+return function(tbl)
+  local version = tbl._format_version
+  if not tbl or not (version == "1.1" or version == "2.1") then
+    return
+  end
+
+  -- migrate top-level routes
+  local routes = table_default(tbl.routes)
+  migrate_routes(routes)
```

Comment:
```suggestion
  migrate_routes(tbl.routes)
```

Reply:
No. This is not what I want to do. The migration should be done with the returned table.